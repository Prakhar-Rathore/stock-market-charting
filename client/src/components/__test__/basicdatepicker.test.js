import React from "react";
import ReactDOM from "react-dom/client";
import BasicDatePicker from "./../basicdatepicker";
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { act } from 'react-dom/test-utils';

it("renders without crashing", async() => {
  const root = ReactDOM.createRoot(document.createElement("div"));
  await act( async () => render(<BasicDatePicker/>));
});

it("renders correctly", () => {
  render(<BasicDatePicker/>)
})

it("matches snapshot", () => {
  const tree = renderer.create(<BasicDatePicker/>).toJSON();
  expect(tree).toMatchSnapshot();
})