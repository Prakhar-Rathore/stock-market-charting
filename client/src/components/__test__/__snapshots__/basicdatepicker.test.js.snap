// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
>
  <label
    className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-e4w4as-MuiFormLabel-root-MuiInputLabel-root"
    data-shrink={false}
    htmlFor="dob"
    id="dob-label"
  >
    Date of Birth
    <span
      aria-hidden={true}
      className="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
    >
      â€‰
      *
    </span>
  </label>
  <div
    className="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-gjbq6i-MuiInputBase-root-MuiFilledInput-root"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      aria-label="Choose date"
      aria-readonly={true}
      autoFocus={true}
      className="MuiFilledInput-input MuiInputBase-input css-10botns-MuiInputBase-input-MuiFilledInput-input"
      id="dob"
      name="dob"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      readOnly={true}
      required={true}
      type="text"
      value=""
    />
  </div>
</div>
`;
